### PR TITLE
Update starknet-rs

### PR DESCRIPTION
**Pull request by Aragar199**
Update starknet-rs to latest version.

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Beerus is using an old version of starknet-rs that does not support Serializaion / Deserialization datatypes that reflect the ecosystem and needs updating.

Issue Number: #364 

# What is the new behavior?

- Update starknet-rs version
- Support Sierra Contract and deprecate Legacy datatype
- Add support for V1 and V2 datatypes

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by phklive** [comment]
I think you have a problem with the associated issue imo it's not #251 right ?

**Comment by Aragar199** [comment]
> I think you have a problem with the associated issue imo it's not #251 right ?

It was a blocker, but not directly related. Created a new issue and linked, thanks!

File: crates/beerus-cli/src/starknet/mod.rs
Comment (Aragar199) [inline crates/beerus-cli/src/starknet/mod.rs:1]: Updated add_declare_transaction to accept compiled_class_hash as an argument to use the latest version.

File: crates/beerus-cli/tests/cli.rs [hunk 1]
```diff
@@ -2867,28 +2868,28 @@ mod test {
         );
 
         let program = vec![];
-        let constructor = vec![ContractEntryPoint {
+        let constructor = vec![LegacyContractEntryPoint {
```
Comment (Aragar199) [inline crates/beerus-cli/tests/cli.rs:2871]: These tests are using the legacy format so just changed to Legacy for now. We can update as part of test coverage efforts.

File: crates/beerus-cli/tests/cli.rs [hunk 2]
```diff
@@ -3609,44 +3610,47 @@ mod test {
...
             external,
             l1_handler,
         };
-        let abi = None;
+        let abi = Default::default();
+        let contract_class_version = Default::default();
 
-        let contract_class: ContractClass = ContractClass {
-            program,
+        let contract_class = SierraContractClass {
```
Comment (Aragar199) [inline crates/beerus-cli/tests/cli.rs:3636]: These needed to be updated as part of updating AddDeclareTransaction to latest version.

File: crates/beerus-cli/src/runner.rs [hunk 3]
```diff
@@ -301,19 +301,19 @@ pub async fn run(beerus: BeerusLightClient, cli: Cli) -> Result<CommandResponse>
 
             StarkNetSubCommands::AddDeclareTransaction {
                 max_fee,
-                version,
```
Comment (Aragar199) [inline crates/beerus-cli/src/runner.rs:304]: "version" argument throughout was removed, did not see it used even in legacy data types so I removed it throughout this PR.

**Comment by Aragar199** [comment]
Changed to draft so I could do some additional testing.

File: crates/beerus-cli/src/runner.rs (hunk 3 above)
Comment (drspacemn) [reply crates/beerus-cli/src/runner.rs:304]: While it looks like `BroadcastedDeclareTransactionV2` does remove version I think we need to leave version almost everywhere else as it is defined in the transaction type:
https://docs.starknet.io/documentation/architecture_and_concepts/Blocks/transactions/#declare_v2

Comment (Aragar199) [reply crates/beerus-cli/src/runner.rs:304]: The language in the version description of the field seemed like it was on the way out and could be removed safely. I can put it back if it's not ready to be removed.

**Comment by phklive** [comment]
lgtm merging

**Pull request merged**